File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.cpp
#include <string>
#include <iostream>
using namespace std;

bool solution(string s)
{
    int count = 0;
    bool answer = true;
    for (char ch : s) {
        if (ch == '(') {
            count++;
        } else if (ch == ')') {
            count--;
        }
        if (count < 0) {
            return false; 
        }
    }
    if (count != 0) answer = false;
    return answer;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        int sum = 0;
        for (int i =0; i < s.length() ; i++) {
            char c = s.charAt(i);
            if (c == '(') {
                sum += 1;
            } else sum -= 1;
            
            if (sum < 0) answer = false;
        }
        if (sum != 0) answer = false;

        return answer;
    }
}